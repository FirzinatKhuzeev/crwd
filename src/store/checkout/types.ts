import { IShopItem } from '../shop/types';

export const ADD_ITEM = 'ADD_ITEM';
export const REMOVE_ITEM = 'REMOVE_ITEM';
export const SHOW_CHECKOUT_MODAL = 'SHOW_CHECKOUT_MODAL';
export const CLEAR_ITEM = 'CLEAR_ITEM';

export interface IShopItemQuantity extends IShopItem {
    quantity: number;
}

export interface ICheckoutState {
    showModal: boolean;
    checkoutItems: IShopItemQuantity[];
}

interface AddItem {
    type: typeof ADD_ITEM;
    payload: IShopItem;
}

interface RemoveItem {
    type: typeof REMOVE_ITEM;
    payload: IShopItemQuantity;
}

interface ClearItem {
    type: typeof CLEAR_ITEM;
    payload: IShopItemQuantity;
}

interface ShowCheckout {
    type: typeof SHOW_CHECKOUT_MODAL;
}

export type CheckoutActions = AddItem | RemoveItem | ClearItem | ShowCheckout;
