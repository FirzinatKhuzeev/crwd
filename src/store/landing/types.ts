export interface IPhoto {
    id: string;
    author: string;
    width: number;
    height: number;
    url: string;
    download_url: string;
}

export interface IPhotoState {
    photos: IPhoto[],
    isFetching: boolean,
}

export const DATA_GATHERING_SUCCESS = "DATA_GATHERING_SUCCESS";
export const DATA_GATHERING_FAILED = "DATA_GATHERING_FAILED";
export const DATA_GATHERING_START = "DATA_GATHERING_START";

export interface IDataGatheringSuccessAction {
    type: typeof DATA_GATHERING_SUCCESS,
    payload: IPhoto[],
    isFetching: false,
}

export interface IDataGatheringFailedAction {
    type: typeof DATA_GATHERING_FAILED,
    isFetching: false,
}

export interface IDataGatheringStartAction {
    type: typeof DATA_GATHERING_START,
    isFetching: true,
}

export type PhotoActions = IDataGatheringStartAction | IDataGatheringSuccessAction | IDataGatheringFailedAction;