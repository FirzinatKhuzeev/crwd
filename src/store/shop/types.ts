export interface IShopCollection {
    id: number;
    title: string;
    category: string;
    items: IShopItem[]
}

export interface IShopItem {
    id: number;
    name: string;
    imageSrc: string;
    price: number;
    description: string;
    size: string[];
    color: string[];
}

export interface IShopState {
    shopCollection: IShopCollection[]
}

export const DATA_GATHERING_SUCCESS = "DATA_GATHERING_SUCCESS";

interface IGetShopDataAction {
    type: typeof DATA_GATHERING_SUCCESS,
    payload: IShopState
}

export type ShopActionTypes = IGetShopDataAction;

