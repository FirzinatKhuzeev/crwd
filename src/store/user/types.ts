export const SIGN_IN_EMAIL_START = 'SIGN_IN_EMAIL_START';
export const SIGN_IN_EMAIL_SUCCEED = 'SIGN_IN_EMAIL_SUCCEED';
export const SIGN_IN_GOOGLE = 'SIGN_IN_GOOGLE';
export const SIGN_OUT = 'SIGN_OUT';
export const SIGN_OUT_START = 'SIGN_OUT_START';

export interface UserState {
    isAuthenticated: boolean | null;
    uuid: string | null;
}

export type UserCredential = {
    email: string;
    password: string;
};

export interface SignInEmailStartAction {
    type: typeof SIGN_IN_EMAIL_START;
    payload: UserCredential;
}

export interface SignInEmailSucceedAction {
    type: typeof SIGN_IN_EMAIL_SUCCEED;
    payload: UserState;
}

export interface SignInGoogleAction {
    type: typeof SIGN_IN_GOOGLE;
}

export interface SignOutAction {
    type: typeof SIGN_OUT;
}

export interface SignOutStartAction {
    type: typeof SIGN_OUT_START;
}

export type UserActions =
    | SignInEmailStartAction
    | SignInEmailSucceedAction
    | SignInGoogleAction
    | SignOutAction
    | SignOutStartAction;
