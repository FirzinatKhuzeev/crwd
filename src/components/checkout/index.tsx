import React from "react";
import { connect } from "react-redux";
import {
    CheckoutPageContainer,
    TotalContainer
} from "./styles";
import { IShopItemQuantity } from "../../store/checkout/types";
import { AppState } from "../../store";
import CheckoutItem from "../checkout-item";

interface ICheckoutProps {
    checkoutItems: IShopItemQuantity[],
    total: number;
}

const Checkout: React.FC<ICheckoutProps> = ({ checkoutItems, total }) => (
    <CheckoutPageContainer>
        {checkoutItems.map((item: IShopItemQuantity) => (
            <CheckoutItem key={item.id} checkoutItem={item} />
        ))}
        {total > 0
            ? (<TotalContainer>
                <span>TOTAL: ${total}</span>
            </TotalContainer>)
            : "Your basket is empty"
        }
    </CheckoutPageContainer>
);

const mapStateToProps = (state: AppState) => {
    return ({
        checkoutItems: state.checkout.checkoutItems,
        total: state.checkout.checkoutItems.reduce((acc, item) => acc + item.price * item.quantity, 0)
    })
};

export default connect(mapStateToProps)(Checkout);
