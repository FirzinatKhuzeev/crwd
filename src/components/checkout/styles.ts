import styled from 'styled-components';
import { CreditCard } from 'styled-icons/boxicons-regular';

export const CheckoutPageContainer = styled.div`
    width: 55%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 50px auto 0;
`;

export const TotalContainer = styled.div`
    margin: 30px 0 30px auto;
`;

export const TotalLabel = styled.span`
    text-transform: uppercase;
    padding-right: 30px;
`;

export const PayButton = styled.button`
    background-color: #2690fb;
    border-radius: 5px;
    color: white;
    height: 35px;
    width: 80px;
    cursor: pointer;
    border: 1px solid transparent;
    font-size: 1rem;
`;

export const CardIcon = styled(CreditCard)``;
