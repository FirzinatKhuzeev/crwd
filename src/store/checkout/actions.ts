import {
    ADD_ITEM,
    REMOVE_ITEM,
    SHOW_CHECKOUT_MODAL,
    ShopItemQuantity,
    CheckoutActions,
    CLEAR_ITEM
} from './types';
import { ShopData } from '../shop/types';

export const addItem = (item: ShopData): CheckoutActions => {
    return {
        type: ADD_ITEM,
        payload: item
    };
};

export const removeItem = (item: ShopItemQuantity): CheckoutActions => {
    return {
        type: REMOVE_ITEM,
        payload: item
    };
};

export const clearItem = (item: ShopItemQuantity): CheckoutActions => {
    return {
        type: CLEAR_ITEM,
        payload: item
    };
};

export const showCheckoutModal = (): CheckoutActions => {
    return {
        type: SHOW_CHECKOUT_MODAL
    };
};
