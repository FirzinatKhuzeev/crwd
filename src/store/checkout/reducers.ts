import {
    ADD_ITEM,
    REMOVE_ITEM,
    SHOW_CHECKOUT_MODAL,
    CheckoutActions,
    ICheckoutState,
    CLEAR_ITEM
} from './types'
import { addItem, removeItem } from './utils';

let initialState: ICheckoutState = {
    showModal: false,
    checkoutItems: []
};

export const checkoutReducer = (
    state = initialState,
    action: CheckoutActions
): ICheckoutState => {

    console.log('ACTION type: ', action.type)
    switch (action.type) {
        case ADD_ITEM:
            return {
                ...state,
                checkoutItems: addItem(state.checkoutItems, action.payload)
            }
        case REMOVE_ITEM:
            return {
                ...state,
                checkoutItems: removeItem(state.checkoutItems, action.payload)
            };
        case CLEAR_ITEM:
            return {
                ...state,
                checkoutItems: state.checkoutItems.filter(
                    cartItem => cartItem.id !== action.payload.id
                )
            };
        case SHOW_CHECKOUT_MODAL:
            return {
                ...state,
                showModal: !state.showModal
            }
        default:
            return state
    }
};