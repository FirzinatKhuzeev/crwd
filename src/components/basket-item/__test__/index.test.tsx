import { shallow } from 'enzyme';
import BasketItem from '../index';
import React from 'react';
import { ShopItemQuantity } from '../../../store/checkout/types';

test('should render Checkout component', () => {
    const mockItem: ShopItemQuantity = {
        id: 1,
        description: 'test image',
        size: [],
        color: [],
        imageSrc: 'http://test.image.com/image.png',
        price: 10,
        name: 'hats',
        quantity: 2
    };
    expect(shallow(<BasketItem checkoutItem={mockItem} />).debug()).toMatchSnapshot();
});
