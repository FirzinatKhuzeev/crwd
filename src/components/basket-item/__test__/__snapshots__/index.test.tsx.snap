// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkout component 1`] = `
"<styled.div>
  <styled.img src=\\"http://test.image.com/image.png\\" alt=\\"item\\" />
  <styled.div>
    <styled.span>
      hats
    </styled.span>
    <styled.span>
      2
       x $
      10
    </styled.span>
  </styled.div>
</styled.div>"
`;
